### PR TITLE
Feature/awesome css

### PR DESCRIPTION
**Pull request by poshul**
## Description

Update doxygen theme to doxygen-awesome. MIT License for theme is compatible with ours.

## Checklist
- [ X ] Make sure that you are listed in the AUTHORS file
- [ X ] Add relevant changes and new features to the CHANGELOG file
- [ X ] I have commented my code, particularly in hard-to-understand areas
- [ X ] New and existing unit tests pass locally with my changes
- [ X ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Cool. Do you have a preview? Can it display everything in a way that is kind of compatible with the old style?

**Comment by poshul** [comment]
Here's a preview. It works with all the content that I have tested it on. I'd definitely welcome more eyes in case I missed something.
image

**Comment by jpfeuffer** [comment]
No, looks promising indeed. I don't think we have a too complex structure anyway. The only thing is:
I had this custom header HTML, that would inject a placeholder for a "Caution: not the latest docs" textbox (which did not work anyway due to a missing JS file).
If we could make that work, it would be great. We should keep that in mind.
It looks like this theme can also have a customizable header though.

**Comment by jpfeuffer** [comment]
https://github.com/OpenMS/OpenMS/issues/6003

File: doc/doxygen/doxygen-awesome-css/.github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: publish
```
Comment (timosachsenberg) [inline doc/doxygen/doxygen-awesome-css/.github/workflows/publish.yaml:1]: are these workflows needed?

Comment (timosachsenberg) [reply doc/doxygen/doxygen-awesome-css/.github/workflows/publish.yaml:1]: I think we are pulling in too many unneeded files. e.g. github actions, docs, img folder need to go. 
maybe also include/MyLibrary (did not check)

**Comment by timosachsenberg** [comment]
there is some strange THIRDPARTY hash. Might be good to set this to THIRDPARTY master

**Comment by timosachsenberg** [comment]
Looks good. Let’s try!

**Review by tapaswenipathak — COMMENTED** [review]
wouldn't it be better to add it as a submodule: https://jothepro.github.io/doxygen-awesome-css/#autotoc_md10?

- it is doxygen theme, that has CSS.
- active development
- nicely supported

why does the front page have different colors than the remaining/default? 

Screenshot 2022-07-21 at 12 31 27 PM

**Comment by poshul** [comment]
> wouldn't it be better to add it as a submodule: https://jothepro.github.io/doxygen-awesome-css/#autotoc_md10?
> 
>     * it is doxygen theme, that has CSS.
> 
>     * active development
> 
>     * nicely supported
> 
> 
> why does the front page have different colors than the remaining/default?
> 
> Screenshot 2022-07-21 at 12 31 27 PM

Color was changed to match our theme colors from our logo. We discussed whether to include it as a submodule or just statically and came to the conclusion that we'd rather not have an added external dependency, and that the overall workload of manually making changes if they are needed was minimal. If you very much disagree I would be open to revisiting the decision.

**Pull request merged**